src/compiler/crystal/syntax/parser.cr

```diff
@@ -333,6 +332,10 @@ module Crystal
               raise "can't change the value of self", location
             end
 
+            if atomic.is_a?(Call) && atomic.name.ends_with?(/[?!]/)
```

Comment:
Using `ends_with?('?')` and `ends_with?('!')` is faster than a regexp
```cr
require "benchmark"

REG = /[?!]/
STR = "123!"

Benchmark.ips do |x|
  x.report("regexp") do
    STR.ends_with? REG
  end

  x.report("chars") do
    STR.ends_with?('?') || STR.ends_with?('!')
  end
end
```
```
regexp 710.41k (  1.41µs) (± 4.25%)  496 B/op  639.10× slower
 chars 454.02M (   2.2ns) (± 2.09%)    0 B/op         fastest
```

Reply:
@j8r thank you for the benchmarking! 
I'll edit the condition to be  `atomic.name.ends_with?('?') || atomic.name.ends_with?('!')`